check_limits.py

```diff
@@ -1,18 +1,77 @@
+import re
+# import googletrans
+from googletrans import Translator
+translator = Translator()
+
 
 def battery_is_ok(temperature, soc, charge_rate):
-  if temperature < 0 or temperature > 45:
-    print('Temperature is out of range!')
-    return False
-  elif soc < 20 or soc > 80:
-    print('State of Charge is out of range!')
-    return False
-  elif charge_rate > 0.8:
-    print('Charge rate is out of range!')
-    return False
+    return (temperature_is_ok(temperature) and soc_is_ok(soc) and charge_rate_is_ok(charge_rate))
+
+
+def temperature_is_ok(temperature):
+    temperature_value = get_value_from_feature(temperature)
+    return is_feature_in_limit(0, 45, temperature_value, 'Temperature')
+
+
+def soc_is_ok(soc):
+    return is_feature_in_limit(20, 80, soc, 'SOC')
+
+
+def charge_rate_is_ok(charge_rate):
+    return is_feature_in_limit(0, 0.8, charge_rate, 'Charge_Rate')
+
+
+def translate_warning(text, language):
+    translated_text = translator.translate(text,  src='en', dest=language)
+    return(translated_text)
+
+
+def print_text(text):
+    print(text)
+    print(translate_warning(text, 'de').text)
+
+
+def get_value_from_feature(feature_value):
+    return int(re.sub('[A-Za-z]',"", feature_value))
+
+
+def get_unit_from_feature(feature_value):
+    return re.sub('[0-9]',"", feature_value)
+
+
+def check_lower_limit(lower_limit, upper_limit, feature_value):
+    if((feature_value <= (lower_limit+(upper_limit*5)/100) and feature_value >= lower_limit)):
+        print_text('Warning: Approaching discharge')
```

Comment:
This function has a strong dependency to print, so we aren't able to test the warning functionality at all.
Consider separating the formation of the string from printing it.
Then it's not only more testable, the string could be used for purposes other than putting on console (like writing to HTML  or pdf)